src/dbnode/namespace/aggregation.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2020  Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package namespace
+
+import (
+	"errors"
+	"fmt"
+	"time"
+)
+
+type aggregationOptions struct {
+	aggregations []Aggregation
+}
+
+// NewAggregationOptions creates new AggregationOptions.
+func NewAggregationOptions() AggregationOptions {
+	return &aggregationOptions{}
+}
+
+func (a *aggregationOptions) SetAggregations(value []Aggregation) AggregationOptions {
+	opts := *a
+	opts.aggregations = value
+	return &opts
+}
+
+func (a *aggregationOptions) Aggregations() []Aggregation {
+	return a.aggregations
+}
+
+func (a *aggregationOptions) Equal(rhs AggregationOptions) bool {
+	if len(a.aggregations) != len(rhs.Aggregations()) {
+		return false
+	}
+
+	for i, agg := range rhs.Aggregations() {
+		if !a.aggregations[i].Equal(agg) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// NewAggregation creates a new Aggregation.
+func NewAggregation(attributes AggregationAttributes, downsampleOptions DownsampleOptions) Aggregation {
+	return Aggregation{
+		Attributes:        attributes,
+		DownsampleOptions: downsampleOptions,
+	}
+}
+
+func (a *Aggregation) Equal(rhs Aggregation) bool {
```

Comment:
Since attributes and downsample options are themselves structs, maybe worth having `Equals` defined on them and call those here?